lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -170,4 +170,85 @@ module.exports = function(
     });
   };
 
+  // This workaround is needed to avoid AngularJS from getting
+  // caught in an infinite digest loop when using `ngRepeat`
+  // over a function that returns a mutable version of an
+  // ImmutableJS object.
+  //
+  // The problem is that every time you call `myImmutableObject.toJS()`
+  // you will get a new object, whose reference is different from
+  // the one you previously got, even if the data is exactly the same.
+  /**
+   * @private
+   * @param {Function} func - function
+   * @param {Function} compare - function
+   * @returns {Function} function
+   */
+  const memoizeImmutableListReference = (func, compare = _.isEqual) => {
```

Comment:
We should definitely about duplicating this, however `lib/shared` it not the place, since this hack is *very* Angular.js specific.

In this comment (https://github.com/resin-io/etcher/pull/1265#discussion_r111381085), I realised the only other occurrence of this hack is also specific to this particular component, so this could be its only home for now.